### ISSUE TITLE
Primitive Color Type for .NET Core

### ISSUE THREAD

**Issue by kendrahavens** [dotnet/runtime#14825]
## Primitive Color Type for .NET Core

The previous discussion of primitives showed that color discussion should expand into a separate issue. 

The primitive discussion was started because many customers wanted to move to the .NET Core, but were blocked by the lack of System.Drawing primitives there. There was much discussion and internal debate over defining Point, Rect, and Size and much of that was fueled by scenarios that you guys cited in the previous primitives issue. (Thank you!) Here we are looking for additional scenarios to discuss the advantages of creating a new Color type for the cross-platform .NET Core Graphics API. Feel free to repost your two cents from the previous primitives issue. Thanks for your input!

Topics of Interest: (This is where providing scenarios would be awesome)
1.  Naming (ColorRgba, sRGB)
2.  Different Color Formats (int8 bits, float 16 bits, float 32 bits, alpha or not) Color4F, Color3F, Color4F
3.  Color Spaces (conversion issues)
4.  Gamma vs Linear
5.  Extension Methods (vector operations: lerp, scaling, conversions: from RGB to HSB and reverse, from vectors)
6.  Semantic parity with CSS
7.  WinRT Compatibility

Limit Scope: KnownColors will not be brought over as they don’t always translate well into the cross-platform world depending on the hardware so let’s focus our discussion on defining the Color object for now.

**Comment by aaronfranke** [comment]
Any update on this? .NET Core 3.1 has been out for awhile, and .NET 5 is intended to be released at the end of this year, so this seems like the appropriate time to add `System.Numerics.Colors`.

I have a branch [here](https://github.com/aaronfranke/corefx/tree/primitive-color), and a closed PR with more information [here](https://github.com/dotnet/corefx/pull/40733). I encourage any .NET Core members to take this code and get it merged and conforming to whatever standards you have, or please let me know what I can do to help or what I need to change about it to get it merged.

**Comment by saucecontrol** [comment]
As was pointed out in your PR that was closed, the first step is a concrete API proposal.  See this doc for the process and examples of how to proceed: https://github.com/dotnet/runtime/blob/master/docs/project/api-review-process.md

**Comment by tannergooding** [comment]
Closing this as I don't think it is super actionable. As indicated by dev's extremely familiar with the area (such as @JimBobSquarePants and @rickbrew) here and on related issues such as <https://github.com/dotnet/runtime/issues/32418>, this space is extremely scenario specific.
Trying to create a general-purpose color type or a set of color types is likely a non-starter and isn't something I can see us adding into the core framework.

To essentially reiterate what was said here: https://github.com/dotnet/runtime/issues/32418#issuecomment-595397329, I think the only actionable thing would be to add some "common algorithms" for specific scenarios where applicable.
For example, the types in `System.Numerics` are oriented around games and multimedia based applications. They are already providing core algorithms similar to the DirectX or OpenGL Math Libraries.

It might be reasonable to then expose some algorithms that operate on a `Vector4` as if it were a `128-bit, 4x float, RGBA` color format. Likewise, it might be reasonable to extend the existing `System.Drawing.Color` or the `System.Windows.Media.Color` types with additional functionality as appropriate.
